OpenEdXMobile/src/main/java/org/edx/mobile/viewModel/InAppPurchasesViewModel.kt

```diff
@@ -33,47 +34,29 @@ class InAppPurchasesViewModel @Inject constructor(
     private val iapAnalytics: InAppPurchasesAnalytics
 ) : ViewModel() {
 
-    private val _productPrice = MutableLiveData<SkuDetails>()
-    val productPrice: LiveData<SkuDetails> = _productPrice
+    private val _productPrice = MutableLiveData<Event<SkuDetails>>()
+    val productPrice: LiveData<Event<SkuDetails>> = _productPrice
 
-    private val _productPurchased = MutableLiveData<Purchase>()
-    val productPurchased: LiveData<Purchase> = _productPurchased
+    private val _launchPurchaseFlow = MutableLiveData<Event<Boolean>>()
+    val launchPurchaseFlow: LiveData<Event<Boolean>> = _launchPurchaseFlow
 
-    private val _executeResponse = MutableLiveData<ExecuteOrderResponse>()
-    val executeResponse: LiveData<ExecuteOrderResponse> = _executeResponse
+    private val _productPurchased = MutableLiveData<Event<IAPFlowData>>()
+    val productPurchased: LiveData<Event<IAPFlowData>> = _productPurchased
 
-    private val _showFullscreenLoaderDialog = MutableLiveData(false)
-    val showFullscreenLoaderDialog: LiveData<Boolean> = _showFullscreenLoaderDialog
+    private val _refreshCourseData = MutableLiveData<IAPFlowData>()
+    val refreshCourseData: LiveData<IAPFlowData> = _refreshCourseData
 
-    private val _purchaseFlowComplete = MutableLiveData(false)
-    val purchaseFlowComplete: LiveData<Boolean> = _purchaseFlowComplete
+    private val _fakeUnfulfilledCompletion = MutableLiveData<Boolean>()
+    val fakeUnfulfilledCompletion: LiveData<Boolean> = _fakeUnfulfilledCompletion
 
-    private val _refreshCourseData = MutableLiveData(false)
-    val refreshCourseData: LiveData<Boolean> = _refreshCourseData
-
-    private val _completedUnfulfilledPurchase = MutableLiveData<Boolean>()
-    val completedUnfulfilledPurchase: LiveData<Boolean> = _completedUnfulfilledPurchase
-
-    private val _showLoader = MutableLiveData(false)
-    val showLoader: LiveData<Boolean> = _showLoader
+    private val _showLoader = MutableLiveData<Event<Boolean>>()
+    val showLoader: LiveData<Event<Boolean>> = _showLoader
 
     private val _errorMessage = MutableLiveData<ErrorMessage?>()
```

Comment:
Please convert the type from `ErrorMessage?` to `Event<ErrorMessage>` and update the relevant code.